codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/client/FluentBuilderGenerator.kt

```diff
@@ -31,10 +33,43 @@ import software.amazon.smithy.rust.codegen.core.smithy.rustType
 import software.amazon.smithy.rust.codegen.core.util.inputShape
 import software.amazon.smithy.rust.codegen.core.util.outputShape
 
+/**
+ * Codegen configuration for the fluent builder generator.
+ *
+ * Null values indicate "use the default", and defaults are oriented towards
+ * the main fluent builders used by the client.
+ *
+ * These can be overridden for secondary fluent builders, such as used by
+ * waiters.
+ */
+interface FluentBuilderConfig {
```

Comment:
This is fine for now but I wonder if it would be nice if things like pagination and waiters could just be customizations and that the base fluent builder didn't need to know about such things (and instead just provided the hooks necessary).

Reply:
The fluent builders do already have a hook to add additional impl methods, but we don't really have a good pattern for config (non-additive) customizations in smithy-rs. That's a problem that still needs to be solved. in addition, there's a difference between "all the fluent builders," and "just these additional fluent builders over here." The customization pattern doesn't really cover that well right now either.

I think config is probably the better approach for now until the customization patterns have been shored up.

Reply:
Ya I wasn't suggesting we do anything about it right now (should have probably marked the comment as "ramblings" or something to make that more clear).